staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go

```diff
@@ -600,18 +600,24 @@ func (q *Quantity) Neg() {
 const int64QuantityExpectedBytes = 18
 
 // String formats the Quantity as a string, caching the result if not calculated.
-// String is an expensive operation and caching this result significantly reduces the cost of
-// normal parse / marshal operations on Quantity.
 func (q *Quantity) String() string {
```

Comment:
IMHO, how about using a `sync.Once` which ignored when marshaled to fix the thread-safety?

btw why r u trying to modify the `Quantity` parallelly? what's the use-case?